code/obj/machinery/manufacturer.dm

```diff
@@ -1,87 +1,88 @@
 #define MAX_QUEUE_LENGTH 20
+#define WIRE_EXTEND 1
+#define WIRE_POWER 2
+#define WIRE_MALF 3
+#define WIRE_SHOCK 4
 
 /obj/machinery/manufacturer
-	name = "Manufacturing Unit"
-	desc = "A standard fabricator unit capable of producing certain items from various materials."
+	name = "manufacturing unit"
+	desc = "A 3D printer-like machine that can construct items from raw materials."
 	icon = 'icons/obj/manufacturer.dmi'
 	icon_state = "fab-general"
-	var/icon_base = "general"
-	density = 1
-	anchored = 1
+	var/icon_base = "general" //! This is used to make icon state changes cleaner by setting it to "fab-[icon_base]"
+	density = TRUE
+	anchored = TRUE
 	mats = 20
+	power_usage = 200
 	req_access = list(access_heads)
 	event_handler_flags = NO_MOUSEDROP_QOL
 	deconstruct_flags = DECON_SCREWDRIVER | DECON_WRENCH | DECON_WELDER | DECON_WIRECUTTERS | DECON_MULTITOOL
 	flags = NOSPLASH | FLUID_SUBMERGE
+
+	// General stuff
 	var/health = 100
+	var/supplemental_desc = null //! appended in get_desc() to the base description, to make subtype definitions cleaner
 	var/mode = "ready"
 	var/error = null
-	var/speed = 3
-	var/repeat = 0
-	var/timeleft = 0
-	var/manual_stop = 0
-	var/panelopen = 0
-	var/powconsumption = 0
-	var/hacked = 0
-	var/malfunction = 0
-	var/electrified = 0
-	var/accept_blueprints = 1
-	var/page = 0 // temporary measure, i want a better UI for this =(
+	var/active_power_consumption = 0 //! How much power is consumed while active? This is determined automatically when the unit starts a production cycle
+	var/panel_open = FALSE
 	var/dismantle_stage = 0
+	var/hacked = FALSE
+	var/malfunction = FALSE
+	var/electrified = 0 //! This is a timer and not a true/false; it's decremented every process() tick
+	var/output_target = null
+	var/list/nearby_turfs = list()
+	var/wires = 15 //! This is a bitflag, for hacking and such
```

Comment:
I don't like this, but I'm not 100% sure what to do about it; ideally the wire defines would just be the bits themselves (e.g. `#define WIRE_EXTEND 1<<0`) and you could sum them to get the initial wire state, but considering this is relying on the global APC wire lists (which seems to be standard for hackable machinery, unfortunately) that would be incredibly verbose. 
Anyways, removing this magic number is a lot more annoying than I'd expect and it's probably out of scope here.
Not requesting a change on this one, just grumbling (feel free to suggest a better idea if you have one)

Reply:
Oh, if BYOND had binary literals I'd just say to use that but HERE WE ARE